Runtime/Device/TouchPanel/AdxTouchPanel.cs

```diff
@@ -113,246 +133,156 @@ public AdxTouchPanel(
-                DebouncedHandleInputChange(TouchPanelZone.B5, HandleB5, currentInput[3]);
-                DebouncedHandleInputChange(TouchPanelZone.B6, HandleB6, currentInput[3]);
-                DebouncedHandleInputChange(TouchPanelZone.B7, HandleB7, currentInput[3]);
-            }
-
-            bool HandleB8(byte input) => handleInputChange(TouchPanelZone.B8, input, 0b00000001);
-            bool HandleC1(byte input) => handleInputChange(TouchPanelZone.C1, input, 0b00000010);
-            bool HandleC2(byte input) => handleInputChange(TouchPanelZone.C2, input, 0b00000100);
-            bool HandleD1(byte input) => handleInputChange(TouchPanelZone.D1, input, 0b00001000);
-            bool HandleD2(byte input) => handleInputChange(TouchPanelZone.D2, input, 0b00010000);
-
-            if (currentInput[4] != _currentState[4])
-            {
-                DebouncedHandleInputChange(TouchPanelZone.B8, HandleB8, currentInput[4]);
-                DebouncedHandleInputChange(TouchPanelZone.C1, HandleC1, currentInput[4]);
-                DebouncedHandleInputChange(TouchPanelZone.C2, HandleC2, currentInput[4]);
-                DebouncedHandleInputChange(TouchPanelZone.D1, HandleD1, currentInput[4]);
-                DebouncedHandleInputChange(TouchPanelZone.D2, HandleD2, currentInput[4]);
-            }
-
-            bool HandleD3(byte input) => handleInputChange(TouchPanelZone.D3, input, 0b00000001);
-            bool HandleD4(byte input) => handleInputChange(TouchPanelZone.D4, input, 0b00000010);
-            bool HandleD5(byte input) => handleInputChange(TouchPanelZone.D5, input, 0b00000100);
-            bool HandleD6(byte input) => handleInputChange(TouchPanelZone.D6, input, 0b00001000);
-            bool HandleD7(byte input) => handleInputChange(TouchPanelZone.D7, input, 0b00010000);
-
-            if (currentInput[5] != _currentState[5])
-            {
-                DebouncedHandleInputChange(TouchPanelZone.D3, HandleD3, currentInput[5]);
-                DebouncedHandleInputChange(TouchPanelZone.D4, HandleD4, currentInput[5]);
-                DebouncedHandleInputChange(TouchPanelZone.D5, HandleD5, currentInput[5]);
-                DebouncedHandleInputChange(TouchPanelZone.D6, HandleD6, currentInput[5]);
-                DebouncedHandleInputChange(TouchPanelZone.D7, HandleD7, currentInput[5]);
-            }
-
-            bool HandleD8(byte input) => handleInputChange(TouchPanelZone.D8, input, 0b00000001);
-            bool HandleE1(byte input) => handleInputChange(TouchPanelZone.E1, input, 0b00000010);
-            bool HandleE2(byte input) => handleInputChange(TouchPanelZone.E2, input, 0b00000100);
-            bool HandleE3(byte input) => handleInputChange(TouchPanelZone.E3, input, 0b00001000);
-            bool HandleE4(byte input) => handleInputChange(TouchPanelZone.E4, input, 0b00010000);
-
-            if (currentInput[6] != _currentState[6])
+            for (var i = 0; i < headIndexs.Length; i++)

```

Comment:
why change this for a for loop the logic that was implemented before should be more efficient as it requires no runtime calculations also with lack of guards you are processing bytes unnecessarily.